server/handlers/user.go

```diff
@@ -103,3 +103,34 @@ func (s *Server) loginImplicitly(w http.ResponseWriter, res *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(user)
 }
+
+func (s *Server) editProfile(w http.ResponseWriter, req *http.Request) {
+
+	verifiedToken := req.Context().Value(ContextKey).(*jwt.Token)
+
+	subject, tokenErr := verifiedToken.Claims.GetSubject()
+	if tokenErr != nil {
+		//not sure what error msg is best here
+		respondWithError(w, "Invalid Token", http.StatusBadRequest)
+		return
+	}
+
+	user := data.UserProfile{}
+	decodeErr := json.NewDecoder(req.Body).Decode(&user)
+	if decodeErr != nil {
+		respondWithError(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
+	updatedUser, updateErr := s.User.EditUser(&user, subject)
+	if updateErr != nil {
+		respondWithError(w, "Internal Server Error: Bad DB Query", http.StatusInternalServerError)
```

Comment:
nitpick: IRL it's not a good idea to tell the client what happened with our database. Since they could use that information to their advantage if trying to attack us. But yes, in this case it is an internal server error, we just don't want to give out TMI lol. Just the status code and a generic error message suffice.